builtin/providers/google/resource_google_project.go

```diff
@@ -58,81 +70,141 @@ func resourceGoogleProject() *schema.Resource {
+			if err != nil {
+				return fmt.Errorf("Error enabling service %q for project %q: %v", v, project.ProjectId, err)
+			}
+			// Wait for the operation to complete
+			waitErr := serviceManagementOperationWait(config, sop, "api to enable")
+			if waitErr != nil {
+				return waitErr
+			}
+		}
 	}
 
-	// Apply the IAM policy if it is set
+	// Apply IAM policy
 	if pString, ok := d.GetOk("policy_data"); ok {
 		// The policy string is just a marshaled cloudresourcemanager.Policy.
 		// Unmarshal it to a struct.
-		var policy cloudresourcemanager.Policy
+		var policy *cloudresourcemanager.Policy
 		if err = json.Unmarshal([]byte(pString.(string)), &policy); err != nil {
 			return err
 		}
 
-		// Retrieve existing IAM policy from project. This will be merged
-		// with the policy defined here.
-		// TODO(evanbrown): Add an 'authoritative' flag that allows policy
-		// in manifest to overwrite existing policy.
-		p, err := getProjectIamPolicy(project, config)
-		if err != nil {
-			return err
-		}
-		log.Printf("[DEBUG] Got existing bindings from project: %#v", p.Bindings)
-
-		// Merge the existing policy bindings with those defined in this manifest.
-		p.Bindings = mergeBindings(append(p.Bindings, policy.Bindings...))
-
-		// Apply the merged policy
-		log.Printf("[DEBUG] Setting new policy for project: %#v", p)
-		_, err = config.clientResourceManager.Projects.SetIamPolicy(project,
-			&cloudresourcemanager.SetIamPolicyRequest{Policy: p}).Do()
+		// Apply the policy
+		log.Printf("[DEBUG] Setting policy for project: %#v", policy)
+		_, err = config.clientResourceManager.Projects.SetIamPolicy(project.ProjectId,
+			&cloudresourcemanager.SetIamPolicyRequest{Policy: policy}).Do()
 
 		if err != nil {
-			return fmt.Errorf("Error applying IAM policy for project %q: %s", project, err)
+			return fmt.Errorf("Error applying IAM policy for project %q. Policy is %+s, error is %s", project.ProjectId, policy, err)
 		}
 	}
 	return nil
 }
 
 func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	project, err := getProject(d, config)
-	if err != nil {
-		return err
-	}
-	d.SetId(project)
-
-	// Confirm the project exists.
-	// TODO(evanbrown): Support project creation
-	p, err := config.clientResourceManager.Projects.Get(project).Do()
+	pid := d.Id()
+	// Read the project
+	p, err := config.clientResourceManager.Projects.Get(pid).Do()
 	if err != nil {
 		if v, ok := err.(*googleapi.Error); ok && v.Code == http.StatusNotFound {
-			return fmt.Errorf("Project %q does not exist. The Google provider does not currently support new project creation.", project)
+			return fmt.Errorf("Project %q does not exist.", pid)
 		}
-		return fmt.Errorf("Error checking project %q: %s", project, err)
+		return fmt.Errorf("Error checking project %q: %s", pid, err)
 	}
 
 	d.Set("number", strconv.FormatInt(int64(p.ProjectNumber), 10))
-	d.Set("name", p.Name)
```

Comment:
It's a required property now, and not computed. Is it best-practice to read-and-set all values?